### DISCUSSION TITLE
Urgent! Can't create branch due to require status check

### DISCUSSION THREAD

**Discussion by kabua** [community/community#112472] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

We are an organization that is trying to do the following:

1. Prevent developers from pushing to `RELEASE/*` without a PR - WORKS.
2. Ensure the CI can build the PR before _merging_ - WORKS.
<img width="742" alt="image" src="https://github.com/community/community/assets/8376646/d291fa62-9add-4572-8284-1f32c3004f8f">
 
3. Only members of the Release team can merge a PR into `RELEASE/*` - WORKS.
4. Only allow the Release team to _create_ branches under `RELEASE`, as in `RELEASE/2.0.0` - **not working**. Why?
Because of this erroneous error...
<img width="452" alt="image" src="https://github.com/community/community/assets/8376646/68c80c8b-406f-4021-8aab-4aa3e3232ae0">

This is a bug, as status checks are for merging only, not for branch creation.

The CI build machine can't build a branch that doesn't exist, therefore, this bug creates a catch-22.

In the mean time, is there a status check that will allow the creation of sub-branches or some other workaround we can try?

**Comment by kabua** [comment]
We have tried disabling the Required status checks from the Branch Protection Rules and enabled it in the Ruleset, still didn't work. 😢

**Comment by kabua** [comment]
Related is this [discussion](https://github.com/orgs/community/discussions/112474).

**Comment by kabua** [comment]
If we add the Release team to the Bypass list
<img width="763" alt="image" src="https://github.com/community/community/assets/8376646/d530f3d1-7c0b-4bbc-bd4d-7b2daf76c092">

Then the release team can create branches like RELEASE/1.0
But now the team can override our PR rules, which isn't good.

<img width="809" alt="image" src="https://github.com/community/community/assets/8376646/33748fc5-5cc4-4c4b-bf81-80e8103873dd">

**Comment by weshaggard** [comment]
I'm hitting the same issue and had to move back to older branch protections for our branch patterns because folks don't have a way to create the branches initially because the required checks are not passing for that commit yet because they haven't ran yet.

**Comment by lee-bcbs** [comment]
I just saw a similar issue right now where we have a Ruleset that targets `release*` for Pull Requests and Status Checks. Creating a new release branch off of main and doing a git push was failing because of 1 of the 3 status checks marked as failure.

I made some changes to the ruleset to test but restored them to previous settings and now it's working again?

<img width="758" alt="image" src="https://github.com/community/community/assets/130779449/c0482f54-cf47-43bf-b564-4eed1389263d">


Just to be clear, Status Checks should *not* fire on pushing a new branch (with no new commits) where the ruleset rules are set like this, correct?

<img width="921" alt="image" src="https://github.com/community/community/assets/130779449/db9a4ce1-fd59-4349-a4f1-3f34e8834d52">

**Comment by Valdermeyder** [comment]
We faced a similar situation with rulesets. We want to allow all team members to create new branches satisfying the `*-release` pattern but we want all updates to them to go through the set of checks like "Require linear history" and "Require status checks to pass"
<img width="438" alt="image" src="https://github.com/community/community/assets/5995043/7cc1bcca-14ac-4949-9d5a-c103a5cc2c5c">
<img width="756" alt="image" src="https://github.com/community/community/assets/5995043/47b1f938-1e33-4b0e-a6b7-9e0c9379cd36">
It works fine for all new changes to the `*-release` branches through the PRs to but when users want to create a branch from the `main` branch they got
<img width="631" alt="image" src="https://github.com/community/community/assets/5995043/8efddbc1-0cb4-4653-a107-aa9d68aed38b">

The workaround for now is to disable these checks in the ruleset and apply them using branch protection rules. The drawback of this is that only admins can see the protection branch rules and other users don't have full visibility on restrictions applied to the branch.

Any plans to fix the behavior?

**Comment by LiuVII** [comment]
We are experiencing the same issue.
Also, this is so huge! It makes rulesets close to completely unusable for branch protections in multi-branch repos.
For example, if, for whatever reason, at the start of repo history, there was a merge commit on `main`, one can't make a new branch out of it if there's a `required_linear_history` rule for the new branch pattern.
GitHub really? Our teams can't create maintenance branches now and we should be switching back to "deprecated" branch protections?!
Given how raw this is, it is ridiculous and outrageous that the ruleset feature doesn't have a `beta` label.

**Comment by nagaraj-ext** [comment]
I too had the same issue when I tried to create new release branch from develop branch, I got the below message
![image](https://github.com/user-attachments/assets/79b63936-9843-49ab-862f-60e4ad15d3b6)

I fixed it by enabling the "Do not require status checks on creation" under require status check ruleset
![image](https://github.com/user-attachments/assets/24edf6e5-90b0-4f26-a128-fedb4028c463)

**Reply by Valdermeyder** [reply]
yes, this fixes the status checks, but not the "Require linear history" condition
